infoblox/infoblox.py

```diff
@@ -1319,6 +1329,114 @@ def get_range(self, network, fields=None, not_found_fail=True):
+                             fields=None, confirm=True):
+        """Create a Fixed Address Record
+        :param ipv4addr: IPv4 Address of object to put on the Record
+        :param mac: Mac Address of object to put on the Record
+        """
+        r_json = self.util.post(
+            'fixedaddress',
+            payload={
+                'mac': mac,
+                'ipv4addr': ipv4addr,
+            },
+            fields=fields,
+            confirm=confirm
+        )
+        return r_json
+
+    def get_fixed_address(self, ipv4addr, mac,
+                          fields=None, not_found_fail=True):
+        """Get a Fixed Address Record
+        :param ipv4addr: IPv4 Address of object to get
+        :param mac: Mac Address of object to get
+        """
+        notFoundText = "Fixed Address not found for IP: %s, MAC: %s" % (ipv4addr, mac)
+        r_json = self.util.get(
+            'fixedaddress',
+            query_params={
+                'mac': mac,
+                'ipv4addr': ipv4addr,
+            },
+            fields=fields,
+            notFoundText=notFoundText,
+            notFoundFail=not_found_fail
+        )
+        return r_json
+
+    def delete_fixed_address(self, ipv4addr, mac, not_found_fail=True):
+        """Delete a Fixed Address Record
+        :param ipv4addr: IPv4 Address of object to delete
+        :param mac: Mac Address of object to delete
+        """
+        ref = self.get_fixed_address(ipv4addr=ipv4addr, mac=mac, not_found_fail=not_found_fail)
+        notFoundText = "Fixed Address not found for ref: %s" % (ref)
+        r_json = self.util.delete_by_ref(
+            ref[0]['_ref'],
+            notFoundText=notFoundText,
+            notFoundFail=not_found_fail
+        )
+        return r_json
+
+    def get_grid(self, name=None, fields=None, not_found_fail=True):
+        """Get a Grid Object
+        :param query_params: Dictionary of searchable fields on Grid object.
+        :param fields: Fields to return from the Grid object
+        """
+        query_params = None
+        if name is not None:
+            query_params = {'name': name}
+        notFoundText = "No grid found."
+        r_json = self.util.get(
+            'grid',
+            query_params=query_params,
+            fields=fields,
+            notFoundText=notFoundText,
+            notFoundFail=not_found_fail
+        )
+        return r_json
+
+    def restart_grid_services(self, payload, name=None):
+        """Restart Grid Services
+        :param name: Name of a Grid object.
+        :param payload: Dictionary of fields used to restart services.
+        """
+        ref = self.get_grid(name=name)
+        uri = '%s?_function=restartservices' % ref[0]['_ref']
+        r_json = self.util.post(
+            uri=uri,
+            payload=payload,
+            fields=None
+        )
+        return r_json
+
+    def get_pending_changes(self, fields=None, notFoundFail=False):
```

Comment:
This only works with `iba_wapi_version` 2.7. Unfortunately the library doesn't allow you to change that dynamically. If I update `iba_wap_version`, it doesn't change the version because `base_url` is set in `__init__`. Maybe we change `base_url` to a property?